tasks/managed/create-pyxis-image/tests/mocks.sh

```diff
@@ -5,9 +5,36 @@ set -exo pipefail
 
 function create_container_image() {
   echo $* >> $(params.dataDir)/mock_create_container_image.txt
-  # The image id is a 4 digit number with leading zeros calculated from the call number,
-  # e.g. 0001, 0002, 0003...
-  echo The image id is $(awk 'END{printf("%04i", NR)}' $(params.dataDir)/mock_create_container_image.txt)
+  
+  # Extract repository name from parameters to determine the line number
+  # This allows us to have a thread-safe way to determine the image id without locking
+  local repository=""
+  local args=("$@")
+  
+  # Parse arguments to extract the --name parameter (repository)
+  for ((i=0; i<${#args[@]}; i++)); do
+    if [[ "${args[i]}" == "--name" && $((i+1)) -lt ${#args[@]} ]]; then
+      repository="${args[i+1]}"
+      break
+    fi
+  done
+  
+  # Find the line number where this repository appears in the file
+  local matching_lines=$(grep -n -- "--name $repository" "$(params.dataDir)/mock_create_container_image.txt")
+  local line_count=$(echo "$matching_lines" | wc -l)
+  
+  # Use the last line number for this repository (handles multi-arch images)
+  # WARNING: this is thread safe as long as the repository name is unique. If you need to
+  # test thread safety, you need to use a different repository name for each request.
```

Comment:
I wanted to call out this warning that I just added to the mock. If anyone is creating new test cases for thread safety, they need to ensure that the repository names are unique.